src/uv-common.c

```diff
@@ -442,3 +442,70 @@ int uv_fs_scandir_next(uv_fs_t* req, uv_dirent_t* ent) {
 
   return 0;
 }
+
+
+static uv_loop_t default_loop_struct;
+static uv_loop_t* default_loop_ptr;
+
+
+uv_loop_t* uv_default_loop(void) {
+  if (default_loop_ptr != NULL)
+    return default_loop_ptr;
+
+  if (uv_loop_init(&default_loop_struct))
+    return NULL;
+
+  default_loop_ptr = &default_loop_struct;
+  return default_loop_ptr;
+}
+
+
+uv_loop_t* uv_loop_new(void) {
+  uv_loop_t* loop;
+
+  loop = (uv_loop_t*)malloc(sizeof(uv_loop_t));
```

Comment:
Unnecessary cast. Also, you can sizeof *loop, resulting in:

~~~~
loop = malloc(sizeof *loop);
~~~~